### PR TITLE
Draft: Fix translation issues

### PR DESCRIPTION
**Pull request by Roy-043**
PR to fix multiple translation issues. Mostly caused by the wrong context.

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

---

### CODE REVIEW

**Comment by yorikvanhavre** [comment]
Thanks... There are still many of those lingering, since the Great Draft Upgrade ;)

**Comment by luzpaz** [comment]
JFYI, there are open Draft related translations issues in https://github.com/FreeCAD/FreeCAD-translations/issues?q=is%3Aissue+is%3Aopen+draft+label%3ADraft

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5370">pipeline status</a> for feature branch [PR_5370](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5370). Pipeline [445061541](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/445061541) was triggered at [63f4ac1](https://github.com/FreeCAD/FreeCAD/pull/5370/commits/63f4ac15fdbb8eda6767c5acf8fae99177fc774b). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
The CI failed. Can you please have a look?

File: src/Mod/Draft/draftguitools/gui_arcs.py [hunk 1]
```diff
@@ -483,15 +482,10 @@ def __init__(self):
...
-        _tip = ("Creates a circular arc by picking 3 points.\n"
-                "CTRL to snap, SHIFT to constrain.")
-
-        d = {'Pixmap': "Draft_Arc_3Points",
-             'MenuText': QT_TRANSLATE_NOOP("Draft_Arc_3Points", _menu),
-             'ToolTip': QT_TRANSLATE_NOOP("Draft_Arc_3Points", _tip),
-             'Accel': 'A,T'}
-        return d
+        return {'Pixmap': "Draft_Arc_3Points",
+                'Accel': "A,T"}
```
Comment (chennes) [inline src/Mod/Draft/draftguitools/gui_arcs.py:486]:
```suggestion
                'Accel': "A,T",
```

File: src/Mod/Draft/draftguitools/gui_arcs.py [hunk 2]
```diff
@@ -483,15 +482,10 @@ def __init__(self):
...
+                'MenuText': QT_TRANSLATE_NOOP("Draft_Arc_3Points", "Arc by 3 points"),
+                'ToolTip': QT_TRANSLATE_NOOP("Draft_Arc_3Points", "Creates a circular arc by picking 3 points.\nCTRL to snap, SHIFT to constrain."),
```
Comment (chennes) [inline src/Mod/Draft/draftguitools/gui_arcs.py:488]:
```suggestion
                'ToolTip': QT_TRANSLATE_NOOP("Draft_Arc_3Points", "Creates a circular arc by picking 3 points.\nCTRL to snap, SHIFT to constrain."),
                }
```

Comment (Roy-043) [reply src/Mod/Draft/draftguitools/gui_arcs.py:488]: Good catch. I should have tested... I'll fix this.

**Comment by Roy-043** [comment]
Note that some translation issues still remain:
https://github.com/FreeCAD/FreeCAD-translations/issues/44
https://github.com/FreeCAD/FreeCAD-translations/issues/45
https://github.com/FreeCAD/FreeCAD-translations/issues/56

This whole translation business is a bit complex, but these issues are even more complex. The GUI elements involved are not 'normal' menus or toolbars. They will have to be addressed in a separate PR.

**Comment by Roy-043** [comment]
The last commit should take care of the remaining translation issues. I have used a dictionary for the tooltips of the snapping commands. For the context I have used `"draft"`, although the command names could have also been used.

Note that in the Draft code both `"Draft"` and `"draft"` are used as the context for translations. The latter more often. This is awkward as the context is case-sensitive. Changing `"Draft"` to `"draft"`, or vice versa, will break translations. I wonder how this can be fixed?

Minor refactoring: The `init_grid_button` function has been removed.

**Comment by Roy-043** [comment]
There seems to be an issue with the `gui_arrays.py` file. AFAICT it is not caused by the edit in this commit.

**Comment by luzpaz** [comment]
@yorikvanhavre after this gets merged can you run a crowdin push/pull (so we can get these translations to see if the fix worked?)

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5370">pipeline status</a> for feature branch [PR_5370](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5370). Pipeline [446837438](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/446837438) was triggered at [843ea92](https://github.com/FreeCAD/FreeCAD/pull/5370/commits/843ea920550e54fb49219d8e5ae15f5294800d87). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by yorikvanhavre** [comment]
Yes, saw the "draft" vs "Draft" context issues already. Not sure what's the best way to solve this or if i is possible without disrupting translations...

**Comment by chennes** [comment]
Does CrowdIn have any way to bulk-push translations up to it, from something like the translation TS files?

**Pull request merged**